### DISCUSSION TITLE
Don't clear the current image if I don't have a place holder and load a new image

### DISCUSSION THREAD

**Discussion by SofiaTejerina** [coil-kt/coil#2461] [Q&A]
Hi!

I'm working on a project where I want to load an animated webp image, but how this image takes some time to be loaded I resolved to first load a static image, from a remote resource, to be shown while the animated webp is being fetched.

My issue is that when you load a new image using coil, the current want is removed until the new one is loaded.

I tried to avoid this by loading the webp on the cache first, passing the placeholder as null, loading the webp on the success event of the static image, among other things I tried, but I didn't succeed.

Do you know if there is some way to do this?

**Comment by colinrtwhite** [comment]
Your best bet is to set the existing image as the placeholder to the new request. Alternatively you could use `rememberAsyncImagePainter` and ignore the painter from the `Loading` state

**Comment by MagenL** [comment]
To solve this issue, in my project, I used an ImageRequest to download the image and save it to the cache, then transferred the image url through a state to the UI when it cached/failed; the result is that there is no downtime/placeholder when it's loading since its already haveve been cached.
